iothub/service/src/Exceptions/IotHubServiceException.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net;
+using System.Runtime.Serialization;
+
+namespace Microsoft.Azure.Devices.Common.Exceptions
+{
+    /// <summary>
+    /// The exception thrown when the client receives an error while communicating with IoT hub service.
+    /// </summary>
+    [Serializable]
+    public class IotHubServiceException : Exception
+    {
+        private const string IsTransientValueSerializationStoreName = "IotHubServiceException-IsTransient";
+        private const string TrackingIdSerializationStoreName = "IoTHubException-TrackingId";
+
+        /// <summary>
+        /// Creates an instance of <see cref="IotHubServiceException"/> with the supplied error message and marks it as non-transient.
+        /// </summary>
+        /// <param name="message">The message that describes the error.</param>
+        /// <param name="innerException">An inner exception, if any.</param>
+        public IotHubServiceException(string message, Exception innerException = default)
+            : base(message, innerException)
+        {
+        }
+
+        /// <summary>
+        /// Creates an instance of <see cref="IotHubServiceException"/> with the supplied error message and tracking Id, and marks it as non-transient.
+        /// </summary>
+        /// <param name="message">The message that describes the error.</param>
+        /// <param name="trackingId">The service returned tracking Id associated with this particular error.</param>
+        public IotHubServiceException(string message, string trackingId)
```

Comment:
Any luck changing the calls to these extra ctors to use the basic one and an initializer list?

Reply:
reworked the ctors here